[charlenetcy] iP

charlenetcy opened the pull request:
# VoidCat :cat:
>"Peering into the void, you realise it was never empty." – Anonymous

VoidCat will remember the unaVOIDable things for you. It's 
- text-based
- easy to learn
- ~~FAST~~ SUPER FAST to use

How to Get Started:
1. Download VoidCat from [here](https://github.com/charlenetcy/ip).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you :wink:

And it is **FREE!**

Features:
- [x] :memo: Task management (todos, deadlines, events)
- [ ] :hourglass: Managing deadlines (coming soon)
- [ ] :spiral_calendar: Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

charlenetcy committed:
Rename,Greet,Exit, LEVEL0

charlenetcy committed:
Added echo, level 1

charlenetcy committed:
Add, list, level-2

charlenetcy committed:
Mark as Done and made test cases for empty values

charlenetcy committed:
todos, events, deadlines, added checks for insufficient args

charlenetcy committed:
Level-4, A-TextUiTesting

charlenetcy committed:
Level-5 Handle Errors

charlenetcy committed:
Level-6 Delete

charlenetcy committed:
add save tasks feature with checks

charlenetcy committed:
Merge branch 'branch-Level-7'

charlenetcy committed:
add dates and times feature for tasks

charlenetcy committed:
Merge branch 'branch-Level-8'

charlenetcy committed:
add more OOP

charlenetcy committed:
Merge branch 'A-More-OOP'

charlenetcy committed:
add packages

charlenetcy committed:
Merge branch 'A-Packages'

naythee169 commented on src/main/java/voidcat/parser/Parser.java:
```diff
@@ -0,0 +1,146 @@
+        String arguments = splitInput.length > 1 ? splitInput[1] : "";
+
+        switch (commandWord.toLowerCase()) {
+            case "bye":
```
case should be same indentation as switch according to java coding standards. May want to consider editing intelliJ configuration to automatically indent correctly?

naythee169 commented on src/main/java/voidcat/storage/Storage.java:
```diff
@@ -0,0 +1,84 @@
+    private Task parseTask(String line) {
+        String[] parts = line.split(" \\| ");
+        switch (parts[0]) {
+            case "T":
```
Same switch case indentation error here

naythee169 commented on src/main/java/voidcat/ui/Ui.java:
```diff
@@ -0,0 +1,90 @@
+        // Display a random greeting
+        System.out.printf(FORMAT, greetings[(int) (Math.random() * greetings.length)]);
+        System.out.println(logo.indent(4));
+        // Display a random assist greeting
```
I like the use of random class to change up the greetings!

naythee169 commented on src/main/java/voidcat/Void.java:
```diff
@@ -0,0 +1,84 @@
+    private TaskList tasks;
+    private Ui ui;
+
+    private String[] greetings = {
```
Missing header comment here.

naythee169 commented on src/main/java/voidcat/parser/Parser.java:
```diff
@@ -0,0 +1,146 @@
+
+public class Parser {
+
+    public void parseAndExecute(String fullCommand, TaskList tasks, Ui ui, Storage storage) throws VoidException, IOException, SecurityException, IllegalArgumentException {
```
Missing header comment here and in multiple other places as well.

naythee169 reviewed (commented):
Overall very good adherence to java coding standards, however take note of the missing header comments as that is a mandatory part of java coding standards.

charlenetcy committed:
Merge branch 'add-gradle-support'

arnav-goel10 commented on src/main/java/voidcat/task/Event.java:
```diff
@@ -0,0 +1,42 @@
+    }
+
+    public Event(String description, String from, String to, int done) {
+        super(description);
+        this.from = LocalDateTime.parse(from, FORMATTER_DATETIME);
+        this.to = LocalDateTime.parse(to, FORMATTER_DATETIME);
```
Instead of repeating the same code, you can just call the other constructor.
```suggestion
        this(description, by); // Call the other constructor
```

arnav-goel10 commented on src/main/java/voidcat/task/Task.java:
Your `Task` class is well put together for handling task descriptions and keeping track of their completion status!

arnav-goel10 commented on src/main/java/voidcat/storage/Storage.java:
```diff
@@ -0,0 +1,84 @@
+                    }
+                }
+                if (tasks.isEmpty()) {
+                    Ui.showMessageAndLines("No saved tasks found yet! voidcat.task.Task list is empty.\n\tStart adding tasks and track them!");
```
Your `Storage` class is handling file operations really well! To make it even better, consider avoiding direct interactions with the `Ui` class. Instead, throw exceptions for any errors and handle them in the `Void` class. This way, your `Storage` class stays focused on its main job i.e. managing files.

charlenetcy committed:
add gradle

charlenetcy committed:
add JUnit tests

charlenetcy committed:
change jar basename

charlenetcy committed:
add JavaDoc

charlenetcy committed:
Merge branch 'branch-A-JavaDoc'

charlenetcy committed:
add coding standards

charlenetcy committed:
Merge branch 'branch-A-CodingStandard'

charlenetcy committed:
add find feature

charlenetcy committed:
solve conflicts in parser and tasklist

charlenetcy committed:
add more javadoc

charlenetcy committed:
add GUI

charlenetcy committed:
add more GUI

charlenetcy committed:
Add assertions and javadocs

Execution will not stop if code has a bug that violates an assumption about how the code should behave.
No help is given to developers to understand the code.

Define assumptions about the program state so that runtime can verify them and if assertion fails, terminate execution sooner for safety.
Documentation helps developers know what the code does and how to use them.

Let's,
*add assertions for task list initialisation and file and folder creation
*add javadocs for clear documentation

charlenetcy committed:
Make happy path prominent and remove dead code

There are some methods where the guard clause comes first and main path is nested deeply.

Keeping the main path prominent and removing dead code improves code quality.

Let's,
*keep the main path un-indented
*remove dead code

charlenetcy committed:
Merge pull request #2 from charlenetcy/branch-A-Assertions

Add assertions and javadocs

charlenetcy committed:
Merge master to code quality

charlenetcy committed:
Merge pull request #3 from charlenetcy/branch-A-CodeQuality

Branch a code quality: Make happy path prominent and remove dead code

charlenetcy committed:
Add better search functionality

charlenetcy committed:
add welcome message and help command

charlenetcy committed:
add User Guide

charlenetcy committed:
update the status icon and cat emojis

charlenetcy committed:
update exception messages to be more consistent

charlenetcy committed:
update images of chatbot

charlenetcy committed:
update user guide to include quick start for users

charlenetcy committed:
update ui image to latest release

charlenetcy committed:
update code to avoid coding style violations using checkstyle
